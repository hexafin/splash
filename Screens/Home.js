import React from 'react';
import {Text, View, Button } from 'react-native';
import Icon from '@expo/vector-icons/Entypo'

 class HomeScreen extends React.Component {
   static navigationOptions = ({ navigation}) => ({
    title: 'Hexa',
    headerRight: <Icon.Button name="new-message" color="white" backgroundColor="#401584" size={28.22} onPress={() => navigation.navigate('Compose')}/>,
    headerLeft: <Icon.Button name="menu" color="white" backgroundColor="#401584" size={25.97} onPress={() => navigation.navigate('DrawerOpen')}/>,
  });
    render() {
       return (
         <View style={{ flex: 1, alignItems: 'center', justifyContent: 'center' }}>
           <Text>Home Screen</Text>
         </View>
       );
     }
  }

export default HomeScreen;
