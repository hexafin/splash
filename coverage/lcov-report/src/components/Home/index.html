<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/Home</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> src/components/Home
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.03% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>27/87</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">7.23% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>6/83</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.03% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>9/29</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">31.03% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>27/87</span>
      </div>
    </div>
    <p class="quiet">
      Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
    </p>
  </div>
  <div class='status-line low'></div>
<div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file medium" data-value="ColoredPayButton.js"><a href="ColoredPayButton.js.html">ColoredPayButton.js</a></td>
	<td data-value="60" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 60%;"></div><div class="cover-empty" style="width:40%;"></div></div></td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	<td data-value="25" class="pct low">25%</td>
	<td data-value="4" class="abs low">1/4</td>
	<td data-value="42.86" class="pct low">42.86%</td>
	<td data-value="7" class="abs low">3/7</td>
	<td data-value="60" class="pct medium">60%</td>
	<td data-value="10" class="abs medium">6/10</td>
	</tr>

<tr>
	<td class="file low" data-value="History.js"><a href="History.js.html">History.js</a></td>
	<td data-value="20.93" class="pic low"><div class="chart"><div class="cover-fill" style="width: 20%;"></div><div class="cover-empty" style="width:80%;"></div></div></td>
	<td data-value="20.93" class="pct low">20.93%</td>
	<td data-value="43" class="abs low">9/43</td>
	<td data-value="8.51" class="pct low">8.51%</td>
	<td data-value="47" class="abs low">4/47</td>
	<td data-value="20" class="pct low">20%</td>
	<td data-value="10" class="abs low">2/10</td>
	<td data-value="20.93" class="pct low">20.93%</td>
	<td data-value="43" class="abs low">9/43</td>
	</tr>

<tr>
	<td class="file low" data-value="Home.js"><a href="Home.js.html">Home.js</a></td>
	<td data-value="32.14" class="pic low"><div class="chart"><div class="cover-fill" style="width: 32%;"></div><div class="cover-empty" style="width:68%;"></div></div></td>
	<td data-value="32.14" class="pct low">32.14%</td>
	<td data-value="28" class="abs low">9/28</td>
	<td data-value="3.13" class="pct low">3.13%</td>
	<td data-value="32" class="abs low">1/32</td>
	<td data-value="44.44" class="pct low">44.44%</td>
	<td data-value="9" class="abs low">4/9</td>
	<td data-value="32.14" class="pct low">32.14%</td>
	<td data-value="28" class="abs low">9/28</td>
	</tr>

<tr>
	<td class="file medium" data-value="index.js"><a href="index.js.html">index.js</a></td>
	<td data-value="50" class="pic medium"><div class="chart"><div class="cover-fill" style="width: 50%;"></div><div class="cover-empty" style="width:50%;"></div></div></td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="0" class="abs high">0/0</td>
	<td data-value="0" class="pct low">0%</td>
	<td data-value="3" class="abs low">0/3</td>
	<td data-value="50" class="pct medium">50%</td>
	<td data-value="6" class="abs medium">3/6</td>
	</tr>

</tbody>
</table>
</div><div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Apr 19 2019 16:14:10 GMT-0400 (Eastern Daylight Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
<script src="../../../block-navigation.js"></script>
</body>
</html>
