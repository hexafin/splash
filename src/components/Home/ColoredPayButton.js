import React, {Component} from "react"
import Svg, {Circle} from "react-native-svg"
import {
	Animated, 
    View,
	Dimensions,
    Image,
    TouchableWithoutFeedback
} from "react-native"
import Color from 'color';
import { defaults, icons } from "../../lib/styles";
const SCREEN_WIDTH = Dimensions.get("window").width
const SCREEN_HEIGHT = Dimensions.get("window").height
import extractBrush from 'react-native-svg/lib/extract/extractBrush';
import {isIphoneX} from "react-native-iphone-x-helper"

let AnimatedCircle = Animated.createAnimatedComponent(Circle)

class ColoredPayButton extends Component {
    componentWillMount() {
        this.scale = new Animated.Value(1)
    }
    componentDidMount() {
        this.props.fillInput.addListener(f => {
            this._component && this._component.setNativeProps({fill: extractBrush(this.props.fill.__getValue())})
        })
    }
    render() {

        return (
            <Animated.View
                style={{
                    position: "absolute",
                    bottom: isIphoneX() ? 40 : 20,
                    width: SCREEN_WIDTH,
                    flexDirection: "row",
                    justifyContent: "center",
                    transform: [{scale: this.scale}]
                }}>
                <TouchableWithoutFeedback
                    onPressIn={() => {
                        Animated.timing(this.scale, {
                            toValue: 1.2,
                            duration: 100
                        }).start()
                    }}
                    onPressOut={() => {
                        Animated.timing(this.scale, {
                            toValue: 1,
                            duration: 100
                        }).start()
                    }}
                    onPress={this.props.onPress}>

                        <View style={{
                            justifyContent: "center",
                            alignItems: "center"
                        }}>
                    
                        	<Svg
                        		width={70} 
                        		height={70} 
                        		viewBox="0 0 70 70">
                			    <AnimatedCircle
                                    ref={component => {this._component = component}}
                                    fill={this.props.fill.__getValue()}
                                    cx="35" cy="35" r="35"/>
                                {/*<Image
                                        style={{
                                            position: "absolute",
                                            top: 18,
                                            left: 16,
                                            width: 35,
                                            height: 35,
                                        }}
                                        source={icons.startPay} resizeMode="contain"/>*/}
                			</Svg>

                            <Image
                                pointerEvents="none"
                                style={{
                                    position: "absolute",
                                    width: 35,
                                    height: 35,
                                    marginRight: 4
                                }}
                                source={icons.startPay} resizeMode="contain"/>

                        </View>
                        
                    
                </TouchableWithoutFeedback>

            </Animated.View>
        );
    }
}
export default ColoredPayButton