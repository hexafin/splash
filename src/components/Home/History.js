import React, {Component} from "react"
import {
	View,
	Text,
	ScrollView,
	Animated,
	StyleSheet,
	Image,
	TouchableWithoutFeedback,
	TouchableOpacity,
	Linking,
	Alert,
	Clipboard,
	Share,
	Dimensions,
	TextInput
} from "react-native"
import { colors } from "../../lib/colors";
import { defaults, icons } from "../../lib/styles";
import { connect } from "react-redux";
import { bindActionCreators } from "redux"
import { isIphoneX } from "react-native-iphone-x-helper"
import ReactNativeHapticFeedback from 'react-native-haptic-feedback'
import TransactionLine from "../universal/TransactionLine"
import { cryptoUnits, decimalToUnits, unitsToDecimal } from '../../lib/cryptos'
import { showViewModal } from '../../redux/modal'

const SCREEN_WIDTH = Dimensions.get("window").width
const SCREEN_HEIGHT = Dimensions.get("window").height

class History extends Component {
	constructor(props) {
		super(props)
		this.state = {
			loading: false,
			height: SCREEN_HEIGHT*3/5,
		}
	}

	componentWillReceiveProps(nextProps) {
		if (nextProps.isLoadingTransactions || nextProps.isLoadingExchangeRates) {
			this.setState({loading: true})
		}
		else {
			this.setState({loading: false})
		}
	}

	shouldComponentUpdate(nextProps, nextState) {
		if (!nextProps.isLoadingExchangeRates && nextProps.exchangeRates.BTC && this.props.exchangeRates.BTC && nextProps.exchangeRates.BTC.USD != this.props.exchangeRates.BTC.USD) {
			return true
		}
		else if (nextProps.currency != this.props.currency) {
			return true
		}
		else if (nextState.loading != this.state.loading) {
			return true
		}
		else if (nextState.modalVisible != this.state.modalVisible || nextState.modalProps != this.state.modalProps) {
			return true
		}
		else if (nextProps.transactions.length != this.props.transactions.length) {
			return true
		}
		else {
			return false
		}
	}

	dynamicHeight(event, stateProp) {
		this.setState({[stateProp]: event.nativeEvent.layout.height})
	}
  
	render() {

		const currencyPrefix = {
			BTC: "BTC ",
			USD: "$"
		}

		return (
			<View style={[styles.history, {
				minHeight: this.state.height,
				paddingBottom: this.props.transactions.length == 0 ? 20 : isIphoneX() ? 140 : 120,
			}]} onLayout={event => this.dynamicHeight(event, "height")}>
				<Text style={styles.sectionTitle}>Your history</Text>
				{this.props.transactions.map(transaction => {
					const satoshiAmount = transaction.type == 'card' ? transaction.amount : transaction.amount.subtotal
					let amount
					if (this.props.exchangeRates.BTC) {
						const rate = decimalToUnits(this.props.exchangeRates.BTC.USD, 'USD')
						amount = this.props.currency == "BTC"
							? unitsToDecimal(satoshiAmount, 'BTC')
							: unitsToDecimal( Math.round((satoshiAmount/cryptoUnits.BTC) * rate), 'USD')
					}
					else {
						amount = 0
					}
					const direction = (transaction.type == "card" || transaction.fromAddress == this.props.bitcoinAddress) ? "to" : "from"

					return (
						<TransactionLine
							key={"transactionLine"+transaction.id}
							transaction={transaction}
							direction={direction}
							amount={currencyPrefix[this.props.currency] + amount}
							loading={this.state.loading}
							onPress={() => {
									this.props.showViewModal({
									  transaction,
									  direction,
					                  address: transaction.type == 'blockchain' ? transaction[direction+'Address'] : null,
				                      exchangeRate: this.props.exchangeRates.BTC["USD"],
				                	})
			             	}}/>
					)
				})}
				{this.props.transactions.length == 0 && <View style={styles.noTransactions}>
					<Text style={styles.noTransactionsText}>
						Send or receive some bitcoin and it{"'"}ll show up here.
					</Text>
					<View style={styles.noTransactionsBottom}>
						<Text style={styles.noTransactionsBottomText}>
							Send some bitcoin!
						</Text>
						<Image resizeMode="contain" style={styles.noTransactionsBottomImage} source={require("../../assets/images/downArrow.png")}/>
					</View>
				</View>}
			</View>
		)
	}
}

const styles = StyleSheet.create({
	noTransactions: {
		flexDirection: "column",
		justifyContent: "space-between",
		alignItems: "center",
		// backgroundColor: colors.gray,
		flex: 1,
	},
	noTransactionsText: {
		color: colors.gray,
		fontSize: 18,
		fontWeight: "400",
		textAlign: "center",
		width: 200,
		marginTop: 36,
	},
	noTransactionsBottom: {
		flexDirection: "column",
		alignItems: "center",
	},
	noTransactionsBottomText: {
		color: colors.gray,
		fontSize: 18,
		fontWeight: "400",
		textAlign: "center",
	},
	noTransactionsBottomImage: {
		width: 12,
		height: 24,
		margin: 12,
	},
	history: {
		padding: 20,
		// backgroundColor: colors.primary
	},
	sectionTitle: {
		color: colors.primaryDarkText,
		fontSize: 20,
		fontWeight: "700"
	},
})

const mapStateToProps = state => {
	return {
		exchangeRates: state.crypto.exchangeRates,
		currency: state.crypto.activeCurrency,
		transactions: state.transactions.transactions,
		isLoadingTransactions: state.transactions.isLoadingTransactions,
		isLoadingExchangeRates: state.crypto.isLoadingExchangeRates,
		bitcoinAddress: state.crypto.wallets.BTC.address,
	}
}

const mapDispatchToProps = dispatch => {
	return bindActionCreators(
		{
			showViewModal
		},
		dispatch
	)
}

export default connect(mapStateToProps, mapDispatchToProps)(History)