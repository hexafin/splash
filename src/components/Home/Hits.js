import React, {Component} from "react"
import { connectInfiniteHits } from 'react-instantsearch/connectors';
import { StyleSheet, FlatList } from 'react-native';
import SplashtagButton from "./SplashtagButton"

const Hits = connectInfiniteHits(({ hits, hasMore, refine, callback }) => {

  /* if there are still results, you can
  call the refine function to load more */
  const onEndReached = function() {
    if (hasMore) {
      refine();
    }
  };

  return (
    <FlatList
      data={hits}
      onEndReached={onEndReached}
      keyExtractor={(item, index) => item.objectID}
      renderItem={({ item }) => {
        return (
          <SplashtagButton
            user={item}
            onPress={() => callback(item)}/>
        )}}
       />       
  );
});

export default Hits