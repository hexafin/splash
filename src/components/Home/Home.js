import React, { Component } from "react";
import {
	View,
	Text,
	ScrollView,
	Animated,
	StyleSheet,
	Image,
	TouchableWithoutFeedback,
	TouchableOpacity,
	Linking,
	Alert,
	Share,
	Dimensions
} from "react-native";
import { colors } from "../../lib/colors";
import { defaults, icons } from "../../lib/styles";
import FCM, { FCMEvent } from "react-native-fcm";
import TransactionLine from "../universal/TransactionLine"
import api from '../../api'
import { isIphoneX } from "react-native-iphone-x-helper"
import moment from "moment"
import { Sentry } from "react-native-sentry";
import LoadingCircle from "../universal/LoadingCircle"
import PayFlow from "./PayFlow"
import ReactNativeHapticFeedback from 'react-native-haptic-feedback';
import { cryptoUnits } from '../../lib/cryptos'

const SCREEN_WIDTH = Dimensions.get("window").width
const SCREEN_HEIGHT = Dimensions.get("window").height


class Home extends Component {
	constructor(props) {
		super(props)
		this.state = {
			currency: "USD",
			balance: null,
			exchangeRate: null,
			transactions: props.transactions,
			loadingExchangeRate: true,
			loadingBalance: true,
			yOffset: new Animated.Value(0),
			pulledToRefresh: false,
			refreshing: false
		}
		this.loadBalance = this.loadBalance.bind(this)
		this.loadExchangeRate = this.loadExchangeRate.bind(this)
		this.refresh = this.refresh.bind(this)
	}

	loadBalance() {
		this.setState(prevState => {
			return {
				...prevState,
				loadingBalance: true,
				balance: null
			}
		})
		api.GetAddressBalance(this.props.bitcoinAddress, this.props.bitcoinNetwork).then(balance => {
			this.setState(prevState => {
				return {
					...prevState,
					balance,
					loadingBalance: false
				}
			})
		}).catch(error => {
			Sentry.messageCapture(error)
			this.setState(prevState => {
				return {
					...prevState,
					balance: null,
					loadingBalance: false
				}
			})
		})
	}

	loadExchangeRate() {
		this.setState(prevState => {
			return {
				...prevState,
				loadingExchangeRate: true,
				exchangeRate: null
			}
		})
		// get exchange rate
        api.GetExchangeRate().then(exchangeRate => {
        	this.setState(prevState => {
        		return {
        			...prevState,
        			exchangeRate,
        			loadingExchangeRate: false
        		}
        	})
        	this.props.UpdateExchangeRate(exchangeRate)
        }).catch(error => {
        	this.setState(prevState => {
        		return {
        			...prevState,
        			exchangeRate: null,
        			loadingExchangeRate: false
        		}
        	})
        })
	}

	refresh() {
		if (!this.state.pulledToRefresh) {
			this.setState(prevState => {
				return {
					...prevState,
					pulledToRefresh: true,
					refreshing: true
				}
			})
			setTimeout(() => {
				this.setState(prevState => {
					return {
						...prevState,
						refreshing: false
					}
				})
			}, 500)
			ReactNativeHapticFeedback.trigger("impactHeavy", true)
			this.loadBalance()
			this.loadExchangeRate()
			this.props.LoadTransactions()
		}
	}

	componentWillReceiveProps(nextProps) {
		this.setState(prevState => {
			return {
				...prevState,
				transactions: nextProps.transactions
			}
		})
		this.loadBalance()
	}

	componentWillMount() {
        if (!this.props.loggedIn) {
            this.props.navigation.navigate("Landing")
        }

		FCM.on(FCMEvent.Notification, async notif => {
			console.log("Notification", notif);
			// reload on notifications
			const {
				transactionId,
				relativeAmount,
				domain,
				relativeCurrency
			} = notif;
			if (transactionId && relativeAmount && domain && relativeCurrency) {
				api.GetExchangeRate().then(exchangeRate => {
					this.props.navigation.navigate("ApproveCardModal", {
						transactionId,
						relativeAmount,
						domain,
						relativeCurrency,
						exchangeRate: exchangeRate[relativeCurrency]
					});
				}).catch(error => {
					// Alert.alert("Could not load exchange rate")
					// TODO: better visual error handling
				})
			}
		});
	}
  
  componentDidMount() {

		this.props.LoadTransactions()

		this.loadExchangeRate()

		// get balance
        this.loadBalance()

	}

	render() {

		const handleBalancePress = () => {
			this.setState(prevState => {
				return {
					...prevState,
					currency: (prevState.currency == "BTC") ? "USD" : "BTC"
				}
			})
		}

		const currencyPrefix = {
			BTC: "BTC ",
			USD: "$"
		}

		let balance = null
		if (this.state.exchangeRate != null && this.state.balance != null) {
			const rate = this.state.exchangeRate[this.state.currency]
			balance = {
				BTC: this.state.balance,
				USD: parseFloat(this.state.balance * rate).toFixed(2)
			}
		}

		let rate = {'USD': 0, 'BTC': 0}
		if (!!this.state.exchangeRate) {
			rate = this.state.exchangeRate
		} else if (!!this.props.exchangeRates) {
			rate = this.props.exchangeRates
		}

		const animatedHeader = {
			opacity: this.state.yOffset.interpolate({
				inputRange: [75, 76, 100, 101],
				outputRange: [0, 0, 1, 1]
			})
		}

		const animatedBalance = {
			transform: [
				{
					scale: this.state.yOffset.interpolate({
						inputRange: [-81, -80, 0, 55, 56],
						outputRange: [1.2, 1.2, 1, 0.8, 0.8]
					})
				},
				{
					translateY: this.state.yOffset.interpolate({
						inputRange: [-1, 0, 53, 54],
						outputRange: [0, 0, -53, -53]
					})
				}
			]
		}

		return (
			<View style={{flex: 1}}>
				<Animated.ScrollView
					scrollEventThrottle={16}
					contentContainerStyle={styles.scrollContainer}
					onScroll={Animated.event(
						[{ nativeEvent: { contentOffset: { y: this.state.yOffset } } }],
						{
							listener: event => {
								const currentY = event.nativeEvent.contentOffset.y
								if (currentY < -80) {
									this.refresh()
								}
								if (currentY >= 0) {
									this.setState(prevState => {
										return {
											...prevState,
											pulledToRefresh: false
										}
									})
								}
							},
							useNativeDriver: true
						}
					)}>
					<View style={styles.container}>
						<Image
							source={require("../../assets/images/headerWaveInverse.png")}
							style={styles.waveInverse}
							resizeMode="contain"/>
						<PayFlow reset={this.state.refreshing}
								 currency={this.state.currency}
								 network={this.props.bitcoinNetwork}
								 exchangeRate={rate}
								 balance={balance}
								 navigation={this.props.navigation}/>
						<View style={styles.history}>
							<Text style={styles.sectionTitle}>Your history</Text>
							{this.state.transactions.map(transaction => {
								const cryptoAmount = transaction.type == 'card' ? transaction.amount/cryptoUnits.BTC : transaction.amount.subtotal/cryptoUnits.BTC
								const amount = this.state.currency == "BTC" ? parseFloat(cryptoAmount*rate[this.state.currency]).toFixed(5) : parseFloat(cryptoAmount*rate[this.state.currency]).toFixed(2)
								const direction = (transaction.type == "card" || typeof transaction.to !== 'undefined') ? "to" : "from"

								return (
									<TransactionLine
										key={"transactionLine"+transaction.id}
										direction={direction}
										amount={currencyPrefix[this.state.currency] + amount}
										date={moment.unix(transaction.timestamp).fromNow()}
										loading={!rate || this.props.isLoadingTransactions}
										title={
											(transaction.type == "card")
											? transaction.domain[0].toUpperCase() + transaction.domain.slice(1)
											: "A bitcoin wallet"
										}
										currency={(transaction.type == 'blockchain' ? transaction.currency : null)}
										onPress={() => {
											this.props.navigation.navigate("ViewTransactionModal", {
												  transaction,
												  direction,
	  						                      address: transaction.type == 'blockchain' ? transaction[direction].address : null,
							                      exchangeRate: rate[this.state.currency],
										  })
										}}
									/>
								)
							})}
						</View>
					</View>
				</Animated.ScrollView>
				<Animated.View style={[styles.header]}/>
				<Animated.View style={[animatedHeader, styles.headerShadow]}/>
				
				<TouchableWithoutFeedback keyboardShouldPersistTaps={"always"} onPress={handleBalancePress}>
					<Animated.View pointerEvents="box-only" style={[animatedBalance, styles.balance]}>
						<Text style={styles.balanceText}>{!(this.state.refreshing || this.state.loadingExchangeRate || this.state.loadingBalance) ? balance[this.state.currency] : " "}</Text>
						<View style={[styles.balanceRefresh, {
							opacity: (this.state.refreshing || this.state.loadingExchangeRate || this.state.loadingBalance) ? 100 : 0
						}]}>
							<LoadingCircle size={30}/>
						</View>
						<View pointerEvents="none" style={styles.balanceCurrencyWrapper}>
							<Image source={icons.refresh} style={styles.refreshIcon}/>
							<Text style={styles.balanceCurrencyText}>{this.state.currency}</Text>
						</View>
					</Animated.View>
				</TouchableWithoutFeedback>
			</View>
		);
	}
}

const styles = StyleSheet.create({
	scrollContainer: {
		marginTop: 0,
		paddingTop: 0,
		position: "relative"
	},
	container: {
		marginTop: 210,
		paddingTop: 0,
		position: "relative",
		backgroundColor: colors.white
	},
	header: {
		flexDirection: "column",
		alignItems: "center",
		justifyContent: "center",
		height: (isIphoneX()) ? 120 : 100,
		width: SCREEN_WIDTH,
		position: "absolute",
		top: 0,
		backgroundColor: colors.primary,
	},
	headerShadow: {
		flexDirection: "column",
		alignItems: "center",
		justifyContent: "center",
		height: (isIphoneX()) ? 120 : 100,
		width: SCREEN_WIDTH,
		position: "absolute",
		top: 0,
		backgroundColor: colors.primary,
		shadowOffset: {
			width: 0,
			height: 10,
		},
		shadowRadius: 12,
		shadowOpacity: 0.2
	},
	balance: {
		flexDirection: "column",
		justifyContent: "center",
		alignItems: "center",
		position: "absolute",
		top: (isIphoneX()) ? 90 : 70,
		width: SCREEN_WIDTH,
	},
	balanceRefresh: {
		position: "absolute",
		width: "100%",
		flexDirection: "row",
		justifyContent: "center",
		top: 0
	},
	balanceText: {
		color: colors.white,
		fontWeight: "600",
		fontSize: 36,
		backgroundColor: "transparent"
	},
	balanceCurrencyWrapper: {
		flexDirection: "row",
		justifyContent: "center",
		alignItems: "center",
		backgroundColor: "transparent"
	},
	balanceCurrencyText: {
		color: "rgba(255,255,255,0.7)",
		fontSize: 16,
		fontWeight: "600",
		marginLeft: 5
	},
	refreshIcon: {
		width: 16,
		height: 16
	},
	waveInverse: {
		width: SCREEN_WIDTH,
		height: 200,
		position: "absolute",
		top: -70
	},
	sectionTitle: {
		color: colors.primaryDarkText,
		fontSize: 20,
		fontWeight: "700"
	},
	history: {
		padding: 20,
		// backgroundColor: colors.white
	},
	sendButton: {
		position: "absolute",
		right: 20,
		bottom: (isIphoneX()) ? 40 : 20,
		justifyContent: 'center',
		alignItems: 'center',
		alignSelf: 'flex-end',
		width: 60,
		height: 60,
		borderRadius: 37.5,
		backgroundColor: '#6364F1',
		shadowOffset: {
			width: 0,
			height: 0,
		},
		shadowOpacity: 0.1,
		shadowRadius: 10,
	},
	sendButtonIcon: {
		width: 30,
		height: 30,
		marginRight: 6,
		marginTop: 2
	}
});

export default Home;
