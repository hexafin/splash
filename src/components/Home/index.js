// container for Home page
import Home from "./Home"
import {connect} from "react-redux"
import {SignIn} from "../../actions/general";
import {bindActionCreators} from "redux";

const mapStatetoProps = state => {
    return {
        person: state.general.person,
        balanceBTC: state.general.balance.btc,
        balanceUSD: (state.general.balance.usd).toFixed(2),
        transactions: state.transactions
    }
}

const mapDispatchToProps = (dispatch) => {
    return bindActionCreators({
        SignIn
    }, dispatch)
}

export default connect(mapStatetoProps, mapDispatchToProps)(Home)
