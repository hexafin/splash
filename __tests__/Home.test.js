import "react-native";
import React from "react";
import Home from "../src/components/Home/Home";
import { History } from "../src/components/Home/History";
import ColoredPayButton from "../src/components/Home/ColoredPayButton";

it("Home renders correctly", () => {
	const snap = shallow(
		<Home
			yOffset={{
				addListener: () => {}
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});

it("History renders correctly", () => {
	const snap = shallow(<History transactions={[]} />);
	expect(snap).toMatchSnapshot();
});

it("Colored Pay Button renders correctly", () => {
	const snap = shallow(
		<ColoredPayButton
			fill={{
				__getValue: () => {}
			}}
			fillInput={{
				addListener: () => {}
			}}
		/>
	);
	expect(snap).toMatchSnapshot();
});
