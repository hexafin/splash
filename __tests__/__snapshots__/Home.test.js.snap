// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colored Pay Button renders correctly 1`] = `
<AnimatedComponent
  style={
    Object {
      "bottom": 40,
      "flexDirection": "row",
      "justifyContent": "center",
      "position": "absolute",
      "transform": Array [
        Object {
          "scale": 1,
        },
      ],
      "width": 750,
    }
  }
>
  <TouchableWithoutFeedback
    onPressIn={[Function]}
    onPressOut={[Function]}
  >
    <Component
      style={
        Object {
          "alignItems": "center",
          "justifyContent": "center",
        }
      }
    >
      <[object Object]
        height={70}
        viewBox="0 0 70 70"
        width={70}
      >
        <AnimatedComponent
          cx="35"
          cy="35"
          r="35"
        />
      </[object Object]>
      <Component
        pointerEvents="none"
        resizeMode="contain"
        source={
          Object {
            "testUri": "../../../src/assets/icons/startPay.png",
          }
        }
        style={
          Object {
            "height": 35,
            "marginRight": 4,
            "position": "absolute",
            "width": 35,
          }
        }
      />
    </Component>
  </TouchableWithoutFeedback>
</AnimatedComponent>
`;

exports[`History renders correctly 1`] = `
<Component
  onLayout={[Function]}
  style={
    Array [
      Object {
        "padding": 20,
      },
      Object {
        "minHeight": 1334,
        "paddingBottom": 20,
      },
    ]
  }
>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#292342",
          "fontSize": 20,
          "fontWeight": "700",
        }
      }
    >
      Your history
    </Component>
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
  >
    <Component
      style={
        Object {
          "color": "#ADADAD",
          "fontSize": 18,
          "fontWeight": "400",
          "lineHeight": 30,
          "marginTop": 36,
          "textAlign": "center",
        }
      }
    >
      No transactions yet.
      

      Here
      '
      s what it could look like...
    </Component>
    <Component
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/images/empty_state.png",
        }
      }
      style={
        Object {
          "height": 200,
          "marginTop": 5,
          "width": 730,
        }
      }
    />
  </Component>
</Component>
`;

exports[`Home renders correctly 1`] = `
<Component
  style={
    Object {
      "height": 1334,
      "width": 750,
    }
  }
>
  <AnimatedComponent
    contentContainerStyle={
      Object {
        "minHeight": 1334,
        "paddingTop": 380,
        "position": "relative",
      }
    }
    onScroll={[Function]}
    scrollEventThrottle={16}
  >
    <Connect(History) />
  </AnimatedComponent>
  <ColoredPayButton
    onPress={[Function]}
  />
</Component>
`;
